File: BinomialCoefficient/C/BinomialCoefficient.c
#include <stdio.h>

long long int factorial(int N)
{
	if(N == 0 || N == 1)
	{
		return 1;
	}
	else if(N > 1)
	{
		long long int result = 1;
		for(int i = 1; i < N + 1; i++)
		{
			result *= i;
		}
		return result;
	}
}

long long int binomialCoefficient(int arr[])
{
	int n = arr[0];
	int k = arr[1];

	long long int result = factorial(n) / (factorial(k) * factorial(n - k));

	return result;
}

int main()
{
	int arrOne[2] = {7, 2};
	int arrTwo[2] = {15, 2};

	printf("arrOne: %lld\n", binomialCoefficient(arrOne));
	printf("arrTwo: %lld\n", binomialCoefficient(arrTwo));



	return 0;
}

File: BinomialCoefficient/Go/BinomialCoefficient.go
package main

import (
	"fmt"
)

func factorial(n int) uint64 {
	
	if(n == 0 || n == 1) {
		
		return 1
	}
	
	var result uint64 = 1
	for i := uint64(2); i < uint64(n) + 1; i++ {

		result *= i
	}

	return result
}

func binomialCoefficient(n int, k int) uint64 {

	return factorial(n) / (factorial(k) * factorial(n - k))
}

func main() {

	fmt.Println(binomialCoefficient(7, 2))
	fmt.Println(binomialCoefficient(15, 2))
}
